src/apps/vpn/mozillavpn.cpp

```diff
@@ -197,15 +198,20 @@ MozillaVPN::MozillaVPN() : App(nullptr), m_private(new MozillaVPNPrivate()) {
   if (!Feature::get(Feature::Feature_webPurchase)->isSupported()) {
     ProductsHandler::createInstance();
   }
+
+#if defined(MZ_ANDROID) || defined(MZ_IOS)
+  PurchaseHandler* purchaseHandler = PurchaseIAPHandler::createInstance();
```

Comment:
Doing this you break the abstraction.

Reply:
There are a bunch of paths to take on this, all with tradeoffs. I'm definitely open to the idea that I didn't pick the best one for our codebase.

If we don't do something like this, I think we need to change the signal that the telemetry file uses to record IAP completion. From the description of this PR:
```
A signal of subscriptionCompleted causes us to record this metric in the telemetry file. It was not being triggered by web purchases, in part because of the two different routes to calling completeActivation:

- For Android/iOS IAP, a completed IAP calls MozillaVPN::subscriptionCompleted, which in turn calls completeActivation.
- Successful web subscriptions call MozillaVPN::completeAuthentication, which also calls completeActivation.
```
Prior to this PR, the web subscription path never called `MozillaVPN::subscriptionCompleted`, but that is what the telemetry file looked for in recording the metrics.

(It seems like `MozillaVPN::completeAuthentication` is called when starting the app when there is already an active subscription, so we can't use it as a sign that we just had a successful web subscription.)

We don't want web purchases to start triggering `MozillaVPN::subscriptionCompleted`. One possibility would be to create a new signal that the telemetry files uses to record the "subscription complete" metrics - and signal it from both the IAP and the web subscription paths. Does that seem cleaner?

Reply:
> We don't want web purchases to start triggering MozillaVPN::subscriptionCompleted

Can you elaborate on why?

Reply:
Apologies for lack of details.

Route for successful web purchases, from `PurchaseWebHandler`:
```
  connect(taskAuthenticate, &TaskAuthenticate::authenticationCompleted, vpn,
          &MozillaVPN::completeAuthentication);
```
Thus, once the web purchase flow starts, the app takes a successful authentication as a sign that the subscription is completed, and continues the flow.

Route for successful IAP, from `MozillaVPN`:
```
connect(purchaseHandler, &PurchaseHandler::subscriptionCompleted, this,
          &MozillaVPN::subscriptionCompleted);
```
WASM, iOS, and Android-specific IAP classes emit a `subscriptionCompleted` to start this path.

As of now, Web-based purchasing flows never emit `subscriptionCompleted`, and thus don't hit `MozillaVPN::subscriptionCompleted`. *Most* of the code within `subscriptionCompleted` is wrapped in `#ifdef`s and wouldn't cause side effects for when run after a successful web-based auth flow - however, both `subscriptionCompleted` and `completeAuthentication` next hit `completeActivation`, and I don't believe we want to run that code twice. 

`completeAuthentication` is called in two situations - when web-based subscriptions succeed, as well as when a typical, already-subscribed user opens the app. So the crux of the problem:
- For web subscriptions, we need to run `completeAuthentication`, which (unless we mess with the login process) will run `completeActivation`.
- For IAP subscriptions, we cannot run `completeAuthentication`, and thus (unless we mess with the login process) it needs to call `completeActivation` on its own path (which currently is `subscriptionCompleted`)